parts/linux/cloud-init/artifacts/ubuntu/cse_install_ubuntu.sh

```diff
@@ -78,6 +82,29 @@ updateAptWithMicrosoftPkg() {
 
 cleanUpGPUDrivers() {
     rm -Rf $GPU_DEST /opt/gpu
+    
+    # Remove cached GPU device plugin downloads on non-GPU nodes
+    if [ -d "/opt/nvidia-device-plugin/downloads" ]; then
+        echo "Removing cached ${K8S_DEVICE_PLUGIN_PKG} downloads from non-GPU node..."
+        rm -rf /opt/nvidia-device-plugin/downloads
+        echo "Cached ${K8S_DEVICE_PLUGIN_PKG} downloads removed from non-GPU node"
+    fi
+}
+
+installNvidiaDevicePluginPkgFromCache() {
+    echo "Installing cached nvidia-device-plugin package..."
+    
+    # Find the cached .deb file with highest version
+    debFile=$(find /opt/nvidia-device-plugin/downloads -name "nvidia-device-plugin*.deb" -type f | sort -V | tail -1)
+    if [ -z "${debFile}" ]; then
+        echo "ERROR: No cached nvidia-device-plugin package found" >&2
+        exit $ERR_GPU_DEVICE_PLUGIN_START_FAIL
+    fi
+    
+    echo "Installing ${debFile}..."
+    dpkg -i "${debFile}" || exit $ERR_APT_INSTALL_TIMEOUT
```

Comment:
we don't have a helper function with retries

we have installDebPackageFromFile and installRPMPackageFromFile

Please have a look at the PR from Lily, which shows all the existing helper functions